x-pack/plugins/observability_onboarding/server/routes/custom_logs/find_latest_observability_onboarding_state.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { SavedObjectsClientContract } from '@kbn/core/server';
+import {
+  OBSERVABILITY_ONBOARDING_STATE_SAVED_OBJECT_TYPE,
+  ObservabilityOnboardingState,
+  SavedObservabilityOnboardingState,
+} from '../../saved_objects/observability_onboarding_status';
+
+export async function findLatestObservabilityOnboardingState({
```

Comment:
it might be enough for now, we can check if the apiKeyId in the client matches the saved object Id to verify that the latest saved obj refers to the current flow. If not, then we can show a message or overwrite or something.